mlir/lib/Dialect/Tensor/Transforms/RewriteAsConstant.cpp

```diff
@@ -45,9 +48,159 @@ struct GenerateToConstant : public OpRewritePattern<GenerateOp> {
   }
 };
 
+/// Rewrite tensor.pack with arith.constant if the pack is writing
+/// to an empty tensor and the destination shape is static.
+struct PackToConstant : OpRewritePattern<tensor::PackOp> {
```

Comment:
Isnt the expectation that these patterns come with a way to control when these patterns get applied?